### ISSUE TITLE
Display which integration is causing the agent to become unhealthy

### ISSUE THREAD

**Issue by jlind23** [elastic/elastic-agent#100]
When an Elastic Agent becomes unhealthy due to an integration, the only way to understand which integration is causing is to remove integrations one by one and/or check logs for a particular error.

Elastic Agent should be able to catch when an integration is failing and must be able to log this within the status command and the diagnostics command.

This is a design tasks between Elastic Agent Data Plane and Elastic Agent Control Plane.

**Comment by jlind23** [comment]
@ph should I keep it in 8.2 or should I rather reconsider it for another release?

**Comment by ph** [comment]
@jlind23 Agree, we need changes in v2 and in the input to be able to better report their state, it's a joined effort from data and control plane.

**Comment by jlind23** [comment]
@cmacknz @kvch if we start working on the input in 8.4. Do you think this can be part of the work achieved?
Whatever the input we take, we should give more meaningful informations when it comes to this particular input state. Thoughts?

cc @ph

**Comment by philippkahr** [comment]
Will the agent go unhealthy only if the integration is not "installable", or will it also catch if an integration is writing error.message fields since it cannot connect to xyz? (e.g. https://github.com/elastic/integrations/issues/3074)

**Comment by jlind23** [comment]
@philippkahr the goal is to catch also when the integration is failing. But then it's going to be up to the integrations developer to properly report statuses.

**Comment by zez3** [comment]
I've got the same abnormal behavior when I change the output. From ES default to a copy if my defaut + dead letters enable

**Comment by prachigupta-qasource** [comment]
Hi @jlind23,

As per the feedback from @kevinlog, we have created test case scenarios for `Endpoint Security` Integration as changes are available for Endpoint Security only

- [[C167461]: Validate that the errors are shown on the 'Endpoint Security' integration dropdown on Agent details page for Mac OS when 'System extension' and 'Full Disk Access' are not granted](https://elastic.testrail.io/index.php?/cases/view/167461)

- [[C167465]: Validate that no error is  shown on the 'Endpoint Security' integration dropdown on Agent details page for Mac OS when 'System extension' and 'Full Disk Access' are granted](https://elastic.testrail.io/index.php?/cases/view/167465)

**Note:** Linux and Windows doesn't require external settings. Therefore, no test cases are created for them.

Further, @joshdover could you please confirm such tags would be shown if the other integrations gets merged.

Thanks!

**Comment by kevinlog** [comment]
@prachigupta-qasource 

Sounds good, thank you!  Also note that @muskangulati-qasource and @harshitgupta-qasource are testing this functionality in Fleet as a part of OLM testing efforts.  I think its a good idea for both teams to be aware of the functionality and have a test plan, but we may also be able to work together and not duplicate too much effort.

cc @manishgupta-qasource

**Comment by prachigupta-qasource** [comment]
Hi Team,

We have executed 02 testcases for this feature under our Fleet Test run at [Fleet 8.4.0-BC3 Feature test plan](https://elastic.testrail.io/index.php?/runs/view/4558&group_by=cases:section_id&group_order=asc) and found that it's **working fine**.

**Build details:**

```
Version: 8.4.0-BC3
Build: 55281
Commit: e42c547d7ab545472fd978383c2c43fa203a5b06
```

Thanks!

**Comment by aleksmaus** [comment]
I pushed the feature-arch-v2 branch to fleet server with initial cut of propagating the detailed status information
https://github.com/elastic/fleet-server/tree/feature-arch-v2

The corresponding agent side PR is posted against the agent branch 
https://github.com/elastic/elastic-agent/pull/916

This should cover the agent/fleet-server side of things for:
https://github.com/elastic/elastic-agent/issues/100

@joshdover @blake.rouse @ph let me know if there is anything else that needs to be addressed related to this feature, can iterate and update things as needed.

The .fleet-agent document looks like the following at the moment:
<img width="993" alt="Screen Shot 2022-08-15 at 12 53 36 PM" src="https://user-images.githubusercontent.com/872351/184683730-d8a72f08-43eb-4ec0-a6b2-b39cf137012b.png">

**Comment by aleksmaus** [comment]
The fleet server and the elastic agent work is complete at the moment, the agent posts the new extended health information to the stack. 

The work is merged to the fleet server and the elastic agent branches respectively:
[fleet server feature-arch-v2](https://github.com/elastic/fleet-server/tree/feature-arch-v2)
[elastic agent feature-arch-v2](https://github.com/elastic/elastic-agent/tree/feature-arch-v2)

It's expected that we will iterate on this feature few more times before release.
@pierrehilbert let me know if you want to keep this open or reassign etc.
